CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/parser/PCAPImportFileParser.java

```diff
@@ -0,0 +1,173 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.plugins.importexport.delimited.parser;
+
+import au.gov.asd.tac.constellation.plugins.parameters.PluginParameters;
+import io.pkts.PacketHandler;
+import io.pkts.Pcap;
+import io.pkts.buffer.Buffer;
+import io.pkts.packet.Packet;
+import io.pkts.packet.IPPacket;
+import io.pkts.packet.TCPPacket;
+import io.pkts.packet.TransportPacket;
+import io.pkts.packet.UDPPacket;
+import io.pkts.protocol.Protocol;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.openide.util.Exceptions;
+import org.openide.util.lookup.ServiceProvider;
+
+
+/**
+ *
+ * @author serpens24
+ */
+@ServiceProvider(service = ImportFileParser.class)
+public class PCAPImportFileParser extends ImportFileParser {
+
+    private static final Logger LOGGER = Logger.getLogger(PCAPImportFileParser.class.getName());    
+    private ArrayList<String[]> allResults = new ArrayList<>();
+    private int frameCounter = 1;
+ 
+    /**
+     * Construct a new JSONImportFileParser with "JSON" label at position 4.
+     */
+    public PCAPImportFileParser() {
+        super("PCAP", 4);
+    }
+        
+    private List<String[]> getResults(final InputSource input, final int limit) throws IOException {
+        try {
+            final String[] headings = {"Frame", "Time", "Source", "Source Port", "Source Type", "Destination", "Destination Port", "Destination Type", "Protocol", "Length"};
+            final ArrayList<String[]> results = new ArrayList<>();
+            
+            // TODO: use classwide allResults to give visibility into nextPacket, is there another way
+            allResults.clear();
+            allResults.add(headings);
+            
+            // Open the supplied PCAP file
+            final Pcap pcap = Pcap.openStream(input.getFile());
+            frameCounter = 1;
+            
+            pcap.loop(new PacketHandler() {
+                @Override
+                public boolean nextPacket(Packet packet) throws IOException {
+                    
+                    IPPacket ipPacket = null;
+                    
+                    SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
+                    Date timestamp = new Date(packet.getArrivalTime() / 1000);
+                    String frame = Integer.toString(frameCounter);
+                    String sourceIP = "";
```

Comment:
Since there are always assigned values, you could use `final String sourceIP;`, etc.

Reply:
As soon as you appy a final to sourceIP you get compilation erros when sourceIP is updated later in the try block.